x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/rules/QueryRuleTests.java

```diff
@@ -152,7 +260,22 @@ public void testToXContentPinnedRuleWithInvalidActions() throws IOException {
         expectThrows(IllegalArgumentException.class, () -> QueryRule.fromXContentBytes(new BytesArray(content), XContentType.JSON));
     }
 
-    public void testApplyRuleWithOneCriteria() {
+    public void testToXContentExcludeRuleWithInvalidActions() throws IOException {
+        String content = XContentHelper.stripWhitespace("""
+            {
+              "rule_id": "my_query_rule",
+              "type": "exclude",
+              "criteria": [
+                { "type": "exact", "metadata": "query_string", "values": ["foo", "bar"] }
+              ],
+              "actions": {
+                  "foo": "bar"
+                }
+            }""");
+        expectThrows(IllegalArgumentException.class, () -> QueryRule.fromXContentBytes(new BytesArray(content), XContentType.JSON));
```

Comment:
Could we test that the returned exception has a helpful message?

Reply:
Maybe i'm misinterpreting, but shouldn't the root cause (from `QueryRule#155`) contain the message "query rule actions must contain only one of either ids or docs"?

Reply:
This is a nit anyways, no changes necessary

Reply:
The API will, but `fromXContent` will say something like "failed to build"